context/src/main/java/io/micronaut/runtime/context/scope/ThreadLocal.java

```diff
@@ -35,4 +36,13 @@
 @Target({ElementType.TYPE, ElementType.METHOD})
 @Scope
 public @interface ThreadLocal {
+    /**
+     * If enabled, track bean life cycle. This means that the bean will be stopped, destroy
+     * listeners will be called etc., when the application context is closed or when the associated
+     * thread terminates. Note that this adds some overhead, so it's off by default.
+     *
+     * @return Whether to enable lifecycle support for this bean
+     * @since 4.9.0
+     */
+    boolean lifecycle() default false;
```

Comment:
is turning this off by default not a breaking change?

Reply:
It only worked for beans that were local to the thread that runs stop() previously, so I think it's more of a breaking change to set it to true than it is to set it to false